src/core/model/types/type.h

```diff
@@ -126,6 +128,10 @@ class Type {
         return !(type_id == +TypeId::kEmpty || type_id == +TypeId::kNull ||
                  type_id == +TypeId::kUndefined || type_id == +TypeId::kMixed);
     }
+
+    virtual Destructor GetDestructor() const {
+        return nullptr;
+    }
```

Comment:
Why not make a `virtual Desturct` method with an empty body in the base class? Given that some types have non-trivial destruction logic and already have method `Destruct` adding `GetDestructor` seems a bit excessive. Also, the problem with `GetDestructor` is a default behavior in the base class, I don't think returning `nullptr` is a good idea, because call to `nullptr` will fail.

Reply:
I thought it would make sense to check if the type was trivially destructible to avoid pointless calls. Checking for `nullptr` does just that